### PR TITLE
common/options,doc: extract formatted desc into .yaml.in 

### PR DESCRIPTION
**Pull request by tchaikov**
this is the third step:

- [x] common: extract options into yaml. finished in #40731
- [x] add an option to the .yaml.in files for generating the corresponding entries in `legacy_config_opts.h`, and expand `y2c.py` to interpret this and generate `legacy_config_opts.h` accordingly. 
- [x] add a sphinx extension to extract and render the basic and advanced options as part of document.
- [ ] use "see_also" to reference other options. 
- [ ] migrate more options under `doc/rados/configuration` to use the `.confval::` directive.
- [ ] split the options groups by its readers / service types. so, for example, mds will be linked against the mds options only. the same applies to all daemons and clients
- [x] use comments in the .yaml.in for the descriptions of some options which don't have desc field yet. // optional.

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/rados/configuration/auth-config-ref.rst [hunk 1]
```diff
@@ -205,34 +176,11 @@ To perform this step manually, execute the following::
...
 .. confval:: keyring
-
-   The path to the keyring file.
-
-  :type: String
-  :required: No
-  :default: ``/etc/ceph/$cluster.$name.keyring,/etc/ceph/$cluster.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin``
-
-
+   :default: /etc/ceph/$cluster.$name.keyring,/etc/ceph/$cluster.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin
```
Comment (jdurgin) [inline doc/rados/configuration/auth-config-ref.rst:180]: this is probably worth a comment in the yaml file, so if we change the default there we know to update the docs. unless you're planning to make this default automatic in the future as well

File: src/common/options/global.yaml.in [hunk 2]
```diff
@@ -7675,11 +7732,17 @@ options:
...
   level: dev
   default: true
+  fmt_desc: Enables direct i/o to the journal. Requires ``journal block
+   align`` set to ``true``.
   with_legacy: true
 - name: journal_aio
   type: bool
   level: dev
   default: true
+  versionchanged: 0.61 Cuttlefish
```
Comment (jdurgin) [inline src/common/options/global.yaml.in:7785]: you could drop the versionchanged support, this is the only place we've used it and it's not important to note a change in cuttlefish anymore.

It seems release notes and different branches in the docs are better places for noting configuration changes

File: doc/_ext/ceph_confval.py [hunk 3]
```diff
@@ -1,12 +1,161 @@
...
+        raise ValueError(f'unknown value: {value}')
+
+
+def readable_duration(value: str, typ: str) -> str:
+    try:
+        if typ == 'float':
+            v = float(value)
+        else:
+            v = int(value)
+        postfix = 'second' if v <= 1 else 'seconds'
```
Comment (jdurgin) [inline doc/_ext/ceph_confval.py:65]: I think it should only use singular for exactly 1, i.e. saying 'that took 0 seconds' is correct

File: doc/_ext/ceph_confval.py [hunk 4]
```diff
@@ -1,12 +1,161 @@
...
+        return f'{v} {postfix}'
+    except ValueError:
+        times = dict(_min=['minute', 'minutes'],
+                     _hr=['hour', 'hours'],
+                     _day=['day', 'days'])
+        for unit, readables in times.items():
+            if value.endswith(unit):
+                v = int(value[:-len(unit)])
+                postfix = readables[0 if v <= 1 else 1]
```
Comment (jdurgin) [inline doc/_ext/ceph_confval.py:74]: only == 1 should be singular

**Review by epuertat — APPROVED** [review]
LGMT! Thanks @tchaikov .

**Review by jdurgin — APPROVED** [review]
looks good in general, left a few minor comments

File: doc/_ext/ceph_confval.py (hunk 3 above)
Comment (tchaikov) [reply doc/_ext/ceph_confval.py]: fixed and repushed.

File: doc/rados/configuration/auth-config-ref.rst (hunk 1 above)
Comment (tchaikov) [reply doc/rados/configuration/auth-config-ref.rst:180]: @jdurin thank you for your review. documented. it's difficult to access cmake variables when building document, and i want to keep the subvar mechinary in cmake. so i have no plan to implement this at this moment.

File: src/common/options/global.yaml.in (hunk 2 above)
Comment (tchaikov) [reply src/common/options/global.yaml.in]: dropped.

File: doc/_ext/ceph_confval.py (hunk 3 above)
Comment (tchaikov) [reply doc/_ext/ceph_confval.py]: thanks. TIL =)

File: doc/rados/configuration/auth-config-ref.rst (hunk 1 above)
Comment (jdurgin) [reply doc/rados/configuration/auth-config-ref.rst:180]: thanks, that makes sense wrt to cmake

**Pull request merged**